src/utils/updateLocaleParam.ts

```diff
@@ -0,0 +1,8 @@
+export const updateLocaleParam = (
+  url: string,
+  currentLocale: string,
+  value: string
+): string => {
+  const post = url.endsWith(`/${currentLocale}`) ? '' : '/';
+  return url.replace(`/${currentLocale}${post}`, `/${value}${post}`);
+};
```

Comment:
Maybe i18n.changeLanguage can be used like previously.

Reply:
fixed